### ISSUE TITLE
Leaderboard: Add ONNX and GGUF to frameworks to allow us to filter based on these  

### ISSUE THREAD

**Issue by KennethEnevoldsen** [embeddings-benchmark/mteb#1622]
See #1579

I believe a way to do this automatically would be to consider Frameworks as a combined set from mteb and those derived from huggingface (we can create a whitelist of tags to look for)

e.g. [Me5-large-instruct](https://huggingface.co/intfloat/multilingual-e5-large-instruct) would have [sentence-transformers, Transformers, ONNX, safetensors]. 
We could potentially rename frameworks to "tags" or split it up into "frameworks" and "formats".

I am unsure if this is valid to fetch during the setup (if we run it once a day I believe it would be) - @x-tabdeveloping WDYT?

**Comment by x-tabdeveloping** [comment]
Great idea! We should inplement it after Christmas!

**Comment by ayush1298** [comment]
We can find all tags of a model using below code snippet: 

```
from huggingface_hub import model_info

info = model_info("intfloat/multilingual-e5-large-instruct")
print(info.tags)
```

But, here the problem is it shows all tags which HF shows at the top of model , basically this thing: 

<img width="1394" height="88" alt="Image" src="https://github.com/user-attachments/assets/0c4a1dca-69b5-4219-9e5a-b7748669c2af" />

So, we can just see what all things from `sentence-transformers, Transformers, ONNX, safetensors, GGUF` are present here or not? 
Are there anything other which we have to check?

**Comment by Samoed** [comment]
I don't think we need this, because we're using only pytorch models and don't have models from onnx and gguf

**Comment by ayush1298** [comment]
> I don't think we need this, because we're using only pytorch models and don't have models from onnx and gguf

Isn't it more about the format that these models support?

**Comment by Samoed** [comment]
We can add additional field to `ModelMeta` for this. But now less models have onnx suport and if they have, they're usually store them in different repository (and GGUF most of the time in different repository too).

**Comment by ayush1298** [comment]
Oh, okay, got it. Then, I think its better to leave this for now(along with adding filed in ModelMeta). Will do both in the future only.

**Comment by KennethEnevoldsen** [comment]
> I don't think we need this, because we're using only pytorch models and don't have models from onnx and gguf

But from a user perspective I might be looking for a GGUF or ONNX compatible model - I think that is still valid metadata.

It seems like (e5) ONNX at least can be stored in the same repo (even if it is often in seperate it indicates support from the model developer)

**Comment by ayush1298** [comment]
> But from a user perspective I might be looking for a GGUF or ONNX compatible model - I think that is still valid metadata.

So, are you suggesting that we should add it as a metadata only? But, then we have to add it that we don't support these models, its just metadata

**Comment by KennethEnevoldsen** [comment]
showing it on the leaderboard does not mean that we support it (it is more about showing that the model provider supports it)

**Comment by ayush1298** [comment]
Okay, then we can add it to similar to model_type filter, 1st adding metadata and then a filter.
I have explained an idea here: https://github.com/embeddings-benchmark/mteb/issues/1622#issuecomment-3681848886

**Comment by KennethEnevoldsen** [comment]
I would just add it to `framework`

**Comment by ayush1298** [comment]
okay, will extend framework only with these

**Comment by ayush1298** [comment]
@KennethEnevoldsen Should I add filter for these in a separate PR or we should not add it for now?

**Comment by KennethEnevoldsen** [comment]
I think it is already there:
https://embeddings-benchmark.github.io/mteb/api/model/?h=framework#mteb.get_model_metas

**Comment by ayush1298** [comment]
@KennethEnevoldsen I think we have to change its implementation. On LB, its currently showing like these only: 

<img width="629" height="117" alt="Image" src="https://github.com/user-attachments/assets/e08da970-8a01-40af-bc99-5d76361452aa" />
